### PR TITLE
[CLI] Implement third-party lib packaging, rollback, and generate

### LINKED ISSUE #1: Python - third party libraries
We just missed shipping this for Usenix Enigma. Jack will update this issue with the plan and updates, it will ship soon

### PR DESCRIPTION
**Pull request by jacknagz**
to @mime-frame 
cc @airbnb/streamalert-maintainers 

size: med

* Package third-party libraries into the AWS Lambda function via the declared `third-party-libs` list in `variables.json`.   Resolves #1 

* Implement `rollback` functionality to decrement the currently running AWS Lambda production version by 1:
`./stream_alert_cli.py lambda rollback --func alert`

* Implement `generate` functionality to generate new Terraform files.  Previously this was only available as part of the `init` process:
`./stream_alert_cli.py terraform generate`.

* Move common helper methods into a new `CLIHelpers` class.

* Fix miscellaneous pylint errors

### CODE REVIEW

File: stream_alert_cli/cli.py [hunk 1]
```diff
@@ -232,7 +243,24 @@ def status(self):
...
+        """Rollback the current production AWS Lambda version by 1
+
+        Notes:
+            Ignores if the production version is $LATEST
+            Only rollsback if published version is greater than 1
+        """
+        clusters = self.config['clusters'].keys()
+        for cluster in clusters:
+            current_vers = self.config['lambda_function_prod_versions'][cluster]
+            if current_vers != '$LATEST':
```
Comment (mime-frame) [inline stream_alert_cli/cli.py]: can `current_vers` ever have the value of '1' or '0' and therefor potentially cause an issue here?

File: stream_alert_cli/helpers.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import logging
```
Comment (mime-frame) [inline stream_alert_cli/helpers.py]: Did you PEP8? I think that would have caught this alpha ordering issue

File: stream_alert_cli/package.py [hunk 3]
```diff
@@ -14,182 +14,212 @@
...
+        with open(package_sha256_path, 'w') as package_sha256_fh:
+            package_sha256_fh.write(package_sha256)
 
-        return pkg_sha256, pkg_sha256_path
+        return package_sha256, package_sha256_path
+
+    def _resolve_third_party(self, temp_package_path):
+        """Install all third-party packages into the deployment package folder
+
+        Note: For Mac OSX/Homebrew users, add the following to ~/.pydistutils.cfg:
```
Comment (mime-frame) [inline stream_alert_cli/package.py]: we should ensure this lands in docs or stdout or something

File: stream_alert_output/encrypted_credentials/pagerduty [hunk 4]
```diff
@@ -0,0 +1 @@
+encrypted pagerduty_url,service_token goes here
```
Comment (mime-frame) [inline stream_alert_output/encrypted_credentials/pagerduty]: for consistency, follow the same text you put int the slack one

File: stream_alert_cli/helpers.py (hunk 2 above)
Comment (jacknagz) [inline stream_alert_cli/helpers.py]: yep, oddly `pylint` likes the order it currently is.

File: stream_alert_cli/cli.py (hunk 1 above)
Comment (mime-frame) [inline stream_alert_cli/cli.py]: chatted offline, we're good here

File: stream_alert_output/encrypted_credentials/pagerduty (hunk 4 above)
Comment (jacknagz) [inline stream_alert_output/encrypted_credentials/pagerduty]: I'm copying in the same format from the docs

**Comment by jacknagz** [comment]
Travis was not responding, verified unit tests locally and they were all fine.

**Pull request merged**